### PR TITLE
Separate test-suite for URLProtocol

### PR DESCRIPTION
**Pull request by mamabusi**

### CODE REVIEW

**Comment by ianpartridge** [comment]
Can we reduce the amount of code duplication by having `TestNSURLProtocol` and `TestURLSession` inherit from a common superclass, which would itself inherit from `XCTestCase`?

**Comment by mamabusi** [comment]
@ianpartridge Yeah, I have moved the server related code to HTTPServer file and have URLSession and URLProtocol tests inherit from there. Please have a look. 
Thanks!

File: TestFoundation/HTTPServer.swift [hunk 1]
```diff
@@ -394,3 +396,39 @@ public class ServerSemaphore {
         dispatchSemaphore.signal()
     }
 }
+
+class ServerSetUp: XCTestCase {
```
Comment (pushkarnk) [inline TestFoundation/HTTPServer.swift]: Can we rename this to something like `LoopbackServerTest`

Comment (ianpartridge) [inline TestFoundation/HTTPServer.swift]: Agreed, `ServerSetUp` is a poor name.

File: TestFoundation/TestNSURLProtocol.swift [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+            let httpResponse = response as! HTTPURLResponse?
+            let responseURL = URL(string: "http://google.com")
+            XCTAssertEqual(responseURL, httpResponse?.url, "Unexpected url")
+            XCTAssertEqual(200, httpResponse!.statusCode, "HTTP response code is not 200")
+        }
+        task.resume()
+        waitForExpectations(timeout: 12)
+    }
+    
+    func test_protocols() {
```
Comment (pushkarnk) [inline TestFoundation/TestNSURLProtocol.swift]: Can we please rename this to something like `test_multipleCustomProtocols`?

File: TestFoundation/HTTPServer.swift (hunk 1 above)
Comment (mamabusi) [inline TestFoundation/HTTPServer.swift]: Done.

File: TestFoundation/TestNSURLProtocol.swift (hunk 2 above)
Comment (mamabusi) [inline TestFoundation/TestNSURLProtocol.swift]: Sure, I have renamed it.

**Comment by mamabusi** [comment]
@pushkarnk Renaming is done. Thanks for the review!

**Comment by pushkarnk** [comment]
@swift-ci test and merge

**Comment by pushkarnk** [comment]
@swift-ci please test and merge

**Pull request merged**